gnomad_qc/v3/create_release/prepare_vcf_data_release.py

```diff
@@ -0,0 +1,974 @@
+    :param format_dict: Dictionary describing MatrixTable entries. Used in header for VCF export.
+    :param inbreeding_coeff_cutoff: InbreedingCoeff hard filter used for variants.
+    :return: Prepared VCF header dictionary.
+    """
+    logger.info("Making FILTER dict for VCF...")
+    filter_dict = make_vcf_filter_dict(
+        hl.eval(t.filtering_model.snv_cutoff.min_score),
+        hl.eval(t.filtering_model.indel_cutoff.min_score),
+        inbreeding_cutoff=inbreeding_coeff_cutoff,
+        variant_qc_filter="AS_VQSR",
+    )
+
+    logger.info("Making INFO dict for VCF...")
+    vcf_info_dict = populate_info_dict(
+        bin_edges=bin_edges,
+        age_hist_data=age_hist_data,
+        subset_list=subset_list,
+        subset_pops=pops,
+    )
+    vcf_info_dict.update({"vep": {"Description": hl.eval(t.vep_csq_header)}})
+
+    # Adjust keys to remove adj tags before exporting to VCF
+    new_vcf_info_dict = {}
+    for i, j in vcf_info_dict.items():
+        i = i.replace("_adj", "")
+        i = i.replace("-", "_")  # VCF 4.3 specs do not allow hyphens in info fields
+        new_vcf_info_dict[i] = j
+
+    header_dict = {
+        "info": new_vcf_info_dict,
+        "filter": filter_dict,
+        "format": format_dict,
+    }
+
+    return header_dict
+
+
+def cleanup_ht_for_vcf_export(
+    ht: hl.Table, drop_freqs: List, drop_hists: Optional[List] = None
+) -> [hl.Table, List[str]]:
+    """
+    Clean up the Table returned by `prepare_vcf_ht` so it is ready to export to VCF with `hl.export_vcf`.
+
+    Specifically:
+        - Drop histograms and frequency entries that are not wanted in the VCF.
+        - Reformat names to remove "adj" and change '-' in info names to '_'.
+        - Adjust types that are incompatible with VCFs using `adjust_vcf_incompatible_types`.
+
+    :param ht: Table returned by `prepare_vcf_ht`.
+    :param drop_freqs: List of frequencies to drop from the VCF export.
+    :param drop_hists: Optional list of histograms to drop from the VCF export.
+    :return: Table ready for export to VCF and a list of fixed row annotations needed for the VCF header check.
+    """
+    # NOTE: For v4 we should just avoid generating any histograms that we will not need rather than dropping them
+    logger.info(
+        "Dropping histograms and frequency entries that are not needed in VCF..."
+    )
+    to_drop = drop_freqs
+    if drop_hists is not None:
+        to_drop.extend(drop_hists)
+
+    ht = ht.annotate(info=ht.info.drop(*to_drop))
+
+    # Reformat names to remove "adj" pre-export
+    # e.g, renaming "AC-adj" to "AC"
+    # All unlabeled frequency information is assumed to be adj
+    logger.info("Dropping 'adj' from info annotations...")
+    row_annots = list(ht.info)
+    new_row_annots = []
+    for x in row_annots:
+        x = x.replace("-adj", "")
+        x = x.replace("-", "_")  # VCF 4.3 specs do not allow hyphens in info fields
```

Comment:
Yeah, we changed because we got reports from users that they were having trouble with the `-` in some programs they were trying to use so we figured we might as well put it in